features/step_definitions/networking.rb

```diff
@@ -1023,3 +1023,74 @@
   end 
   logger.info "The multicast is enable in the #{project_name} project"
 end
+
+Given /^I get the ptp logs of the "([^"]*)" node since "(.+)" ago$/ do | node_name, duration |
+  ensure_admin_tagged
+  node_name ||= node.name
+
+  # Only return logs newer than a relative duration like 5s, 2m, or 3h.
+  ptp_pod = BushSlicer::Pod.get_labeled("app=linuxptp-daemon", project: project("openshift-ptp", switch: false), user: admin) { |pod, hash|
+    pod.node_name == node_name
+  }.first
+  @result = admin.cli_exec(:logs, resource_name: ptp_pod.name, n: "openshift-ptp", since: duration)
+end
+
+Given /^the ptp operator is running well$/ do
+  ensure_admin_tagged
+  step %Q/I switch to cluster admin pseudo user/
+  step %Q/evaluation of `cluster_version('version').channel.split('-')[1]` is stored in the :ocp_cluster_version clipboard/
+
+  unless project('openshift-ptp').exists?
+    if cb.ocp_cluster_version.include? "4.[34]."
+      # create 4.3/4.4 namespaces
+      step %Q{I obtain test data file "networking/ptp/namespace/43/ns43.yaml"}
+      step %Q/I run the :create admin command with:/,table(%{
+        | f | ns43.yaml |
+      })
+      step %Q{the step should succeed}
+    else
+      # Create namespace for 4.5 and later
+      step %Q{I obtain test data file "networking/ptp/namespace/45/ns45.yaml"}
+      step %Q/I run the :create admin command with:/,table(%{
+        | f | ns45.yaml |
+      })
+      step %Q{the step should succeed}
+    end
+  end
```

Comment:
```
  ensure_admin_tagged
  step %Q/I switch to cluster admin pseudo user/
  step %Q/evaluation of `cluster_version('version').channel.split('-')[1]` is stored in the :ocp_cluster_version clipboard/

  unless project('openshift-ptp').exists?
    if cb.ocp_cluster_version.include? "4.[34]."
      # create 4.3/4.4 namespaces
      step %Q{I obtain test data file "networking/ptp/namespace/43/ns43.yaml"}
      step %Q/I run the :create admin command with:/,table(%{
        | f | ns43.yaml |
      })
      step %Q{the step should succeed}
    else
      # Create namespace for 4.5 and later
      step %Q{I obtain test data file "networking/ptp/namespace/45/ns45.yaml"}
      step %Q/I run the :create admin command with:/,table(%{
        | f | ns45.yaml |
      })
      step %Q{the step should succeed}
    end
  end
```
can be shorted as
```
  ensure_admin_tagged
  step %Q/I switch to cluster admin pseudo user/

  unless project('openshift-ptp').exists?
    if env.version_eq("4.3", user: user) || env.version_eq("4.4", user: user)
      ns = ns43.yaml
    elsif env.version_ge("4.5", user: user)
      ns = ns45.yaml
    fi
    step %Q{I obtain test data file "networking/ptp/namespace/#{ns}"}
    step %Q/I run the :create admin command with:/,table(%{
      | f | #{ns} |
    })
    step %Q{the step should succeed}
  end
```

Reply:
@liangxia Thanks for optimizing codes, fixed it.